src/budget/CategoryRow.js

```diff
@@ -1,15 +1,129 @@
 import React from 'react';
+import { PropTypes } from 'prop-types';
+import { useSelector } from 'react-redux';
+import Button from 'react-bootstrap/Button';
+import ProgressBar from 'react-bootstrap/ProgressBar';
 
 /**
  * Shows the category title, a progress bar, how much was spent out of the limit
  * and a button to view more details
+ * expects to go in a thing with a variable of whatever
  *
  * @component
  * @example
  *
+ * PUT AN EXAMPLE HERE
  */
-export default function CategoryRow() {
+
+// Need to do:
+// send data in details link
+//  ?what happens on the details page?
+//   <Route path="budget/:categoryId" element={<CategoryDetailsPage />} />
+// ?how should I be getting the time period?
+//   Hard code for now until there's a means to get the desired period.
+// test edge cases
+// edge cases:
+// underspent (return): $0 out of $50 (crossed out) $90 with empty bar
+// overspent: $120 out of $100 with full bar
+
+function CategoryRow({ category }) {
+  const startDate = new Date('06-01-2021');
+  const endDate = new Date('06-30-2021');
+  // check out memoization
+  const amountSpent = useSelector((state) => state.transactions.entities.filter(
+    (transaction) => transaction.categoryId === category.id
+      && transaction.date >= startDate && transaction.date <= endDate,
+  ).reduce((accumulator, currentValue) => accumulator + currentValue.amount, 0).toFixed(2)) * -1;
+
+  const varianceToPercentage = (amountSpent / category.target) * 100;
+
+  let barColor = '';
+  let message = (
+    <span>
+      $
+      {amountSpent.toFixed(2)}
```

Comment:
Since you're very frequently formatting dates by putting a $ and rounding it to two decimal points, it might make sense to make yourself a little helper function (maybe named formatAsMoney) that formats numbers like this. Then you can just call it everywhere you need to format like this.

Reply:
This sounds good and useful for everyone. Is this something I can put in a separate file so everyone can use it?

Reply:
Ooh, that's a fantastic idea! And you know what, I just realized there is a formatMoney function in the utilities/index.js file. If that doesn't work for your purposes, then you could make another function in there (maybe named something like formatMoneySimple)